### PR TITLE
fix: user selects none; location disappears correctly

### PR DESCRIPTION
**Pull request by abp6318**
### Important
Make sure you have your opendata-ui set to https://github.com/ArcGIS/opendata-ui/pull/12378 until it merges, I added a new translation this morning.

1. Description: 
- If a user selects "none" in the location picker, it no longer appears in the /about view
- Location picker options in workspace content now properly shows two options
  - This will be further built off of in future issues and PRs that allow for a bounding box option as discussed with Tom, Dave, and Ben on Aug 10, 2023

1. Instructions for testing:
- Clone this PR and PR with updated translations: https://github.com/ArcGIS/opendata-ui/pull/12378 
- Copy the hubjs dist folder over
- `yarn hc start` -- switch back and forth between "none" and "Item's Current Extent" in content's location picker
- Double check that I haven't changed other workspace entities like projects

1. Closes Issues: #7208

1. [ ] Updated meaningful TSDoc to methods including Parameters and Returns, see [Documentation Guide](https://esri.github.io/hub-components/storybook/?path=/story/guides-documentation--page)

1. [x] used semantic commit messages
  
1. [x] PR title follows semantic commit format (**CRITICAL** if the title is not in a semantic format, the release automation will not run!)

1. [ ] updated `peerDependencies` as needed. **CRITICAL** our automated release system can **not** be counted on to update `peerDependencies` so we _must_ do it manually in our PRs when needed. See the [updating peerDependencies](/RELEASE.md#Updating-peerDependencies) section of the release instructions for more details.

### CODE REVIEW

File: packages/common/src/content/edit.ts [hunk 1]
```diff
@@ -104,6 +105,12 @@ export async function updateContent(
   // we are not attempting to handle "concurrent edit" conflict resolution
   // but this is where we would apply that sort of logic
   const modelToUpdate = mapper.entityToStore(content, model);
+
+  // prevent map from displaying when boundary is 'none'
+  const itemType = getProp(modelToUpdate, "item.properties.location.type");
+  modelToUpdate.item.properties.boundary =
+    itemType === "none" ? "none" : "item";
```
Comment (tomwayson) [inline packages/common/src/content/edit.ts:112]: `locationType` is more accurate

```suggestion
  const locationType = getProp(modelToUpdate, "item.properties.location.type");
  modelToUpdate.item.properties.boundary =
    locationType === "none" ? "none" : "item";
```

File: packages/common/test/content/edit.test.ts [hunk 2]
```diff
@@ -70,6 +70,52 @@ describe("content editing:", () => {
...
+        createdDate: new Date(1595878748000),
+        createdDateSource: "item.created",
+        updatedDate: new Date(1595878750000),
+        updatedDateSource: "item.modified",
+        thumbnailUrl: "",
+        permissions: [],
+        schemaVersion: 1,
+        canEdit: false,
+        canDelete: false,
+        location: { type: "item" },
```
Comment (tomwayson) [inline packages/common/test/content/edit.test.ts:118]: we prob want to add expectations on `modelToUpdate.item.properties.boundary` for this test to be meaningful

**Review by tomwayson — COMMENTED** [review]
As we discussed, please add a commit where get a distinct location editor config for content.

File: packages/common/src/core/schemas/internal/getLocationOptions.ts [hunk 3]
```diff
@@ -22,9 +23,69 @@ export async function getLocationOptions(
...
+  ).map((option) => {
+    // If this is a new entity, select the custom option by default
+    if (!entity.id && option.location.type === "custom") {
+      option.selected = true;
+    } else if (entity.id && !location && option.location.type === "none") {
+      option.selected = true;
+    } else if (location?.type === option.location.type) {
+      option.location = location;
+      option.selected = true;
+    }
```
Comment (sonofflynn89) [inline packages/common/src/core/schemas/internal/getLocationOptions.ts:74]: It took me a while to figure out why you had these if statements. I think my biggest hangup is the dual purpose of checking for the option's type AND seeing if a fallback is necessary. The re-assigning of `option.location` also is oddly placed, since the `custom` option is the only one with a location to begin with.

Don't take too long, but take a crack at simplifying this. Try to break up these 3 steps:
- Determining which location type is selected, including fallbacks
- Setting the `custom` option's `location` field
- Actually marking an option as selected

**Review by sonofflynn89 — COMMENTED** [review]
Great work! Just a couple comments

**Review by sonofflynn89 — APPROVED** [review]
Great work with all this Aaron! Thank you so much for taking this on 🙏

File: packages/common/src/core/schemas/internal/getLocationOptions.ts (hunk 3 above)
Comment (tomwayson) [reply packages/common/src/core/schemas/internal/getLocationOptions.ts:80]: I agree that coupling those operations do make this a little hard to follow, but there is a bigger issue that needs to be resolved first, which is that we need to base the options and selected state on the item's extent and `properties.boundary` and **not**  the org's extent and `properties.location`.

**Review by tomwayson — CHANGES_REQUESTED** [review]
All in all this is good, we just need to update the logic to work with existing content items. Let's pair on this.

File: packages/common/src/core/schemas/internal/getLocationOptions.ts [hunk 4]
```diff
@@ -37,50 +48,33 @@ export async function getLocationOptions(
...
-  );
-  const location: IHubLocation = entity.location;
-  return (
-    [
-      {
-        label: "{{shared.fields.location.none:translate}}",
-        location: { type: "none" },
+  const defaultExtent: IExtent = getItemExtent(entity.extent);
+  const defaultExtentIsBBox = isBBox(entity.extent);
+  const boundary = entity.boundary;
```
Comment (tomwayson) [inline packages/common/src/core/schemas/internal/getLocationOptions.ts:55]: is `entity.boundary` a thing? I was expecting that you'd have to reach into `item.properties` to get `boundary`?

**Review by tomwayson — APPROVED** [review]
Good work. As long as content entities have a `.boundary` the logic in this is sound.
